### PR TITLE
add code comments

### PR DESCRIPTION
**Pull request by skothari-tibco**

### CODE REVIEW

**Comment by abramvandergeest** [comment]
I have added Matt and Frank for the review of documents to make sure it is what we are looking for,  The issue this is addressing is Issue https://github.com/project-flogo/catalystml-flogo/issues/62

**Review by abramvandergeest — APPROVED** [review]
LGTM.  Everything that is in it looks good.  I think @fm-tibco and @mellistibco should see if they think anything more should be in the documentation.  Thanks for doing this Samip!

**Review by mellistibco — APPROVED** [review]

File: action/api/api.go [hunk 1]
```diff
@@ -8,16 +8,18 @@ import (
 
 type Option func(*action.Config)
 
+//Initiliaize the action.
```
Comment (fm-tibco) [inline action/api/api.go]: Take a look at https://blog.golang.org/godoc-documenting-go-code.  We should probably follow their conventions for the go doc.

It states:

> Notice this comment is a complete sentence that begins with the name of the element it describes. This important convention allows us to generate documentation in a variety of formats, from plain text to HTML to UNIX man pages, and makes it read better when tools truncate it for brevity, such as when they extract the first line or sentence.
>

**Review by fm-tibco — CHANGES_REQUESTED** [review]

**Review by abramvandergeest — APPROVED** [review]
LGTM

**Review by fm-tibco — APPROVED** [review]

**Pull request merged**